services/github/github-issue-detail.service.js

```diff
@@ -1,27 +1,142 @@
+  errorMessagesFor,
+  stateColor,
+  commentsColor,
 } = require('./github-helpers')
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class GithubIssueDetail extends LegacyService {
+const getDefaultLabel = ({ json }) => {
+  if (json.pull_request) {
+    return `pull request ${json.number}`
+  }
+
+  return `issue ${json.number}`
+}
+
+const commonSchemaFields = {
+  number: nonNegativeInteger,
+  pull_request: Joi.any(),
+}
+
+const stateMap = {
+  schema: Joi.object({
+    ...commonSchemaFields,
+    state: Joi.string()
+      .allow('open', 'closed')
+      .required(),
+  }).required(),
+  transform: ({ json }) => json.state,
+  render: ({ value, json }) => ({
+    color: stateColor(value),
+    label: getDefaultLabel({ json }),
+    message: value,
+  }),
+}
+
+const titleMap = {
+  schema: Joi.object({
+    ...commonSchemaFields,
+    title: Joi.string().required(),
+  }).required(),
+  transform: ({ json }) => json.title,
+  render: ({ value, json }) => ({
+    label: getDefaultLabel({ json }),
+    message: value,
+  }),
+}
+
+const authorMap = {
+  schema: Joi.object({
+    ...commonSchemaFields,
+    user: Joi.object({
+      login: Joi.string().required(),
+    }).required(),
+  }).required(),
+  transform: ({ json }) => json.user.login,
+  render: ({ value }) => ({
+    label: 'author',
+    message: value,
+  }),
+}
+
+const labelMap = {
+  schema: Joi.object({
+    ...commonSchemaFields,
+    labels: Joi.array()
+      .items(
+        Joi.object({
+          name: Joi.string().required(),
+          color: Joi.string().required(),
+        })
+      )
+      .required(),
+  }).required(),
+  transform: ({ json }) => {
+    if (json.labels.length === 0) {
+      throw new InvalidResponse({ prettyMessage: 'no labels found' })
+    }
+    return json.labels.map(l => l.name).join(' | ')
+  },
+  render: ({ value, json }) => {
+    let color
+    if (json.labels.length === 1) {
+      color = json.labels[0].color
+    }
+    return {
+      color,
+      label: 'label',
+      message: value,
+    }
+  },
+}
+
+const commentsMap = {
+  schema: Joi.object({
+    ...commonSchemaFields,
+    comments: nonNegativeInteger,
+  }).required(),
+  transform: ({ json }) => json.comments,
+  render: ({ value }) => ({
+    color: commentsColor(value),
+    label: 'comments',
+    message: metric(value),
+  }),
+}
+
+const ageUpdateMap = {
+  schema: Joi.object({
+    ...commonSchemaFields,
+    created_at: Joi.date().required(),
+    updated_at: Joi.date().required(),
+  }).required(),
+  transform: ({ json, which }) =>
+    which === 'age' ? json.created_at : json.updated_at,
+  render: ({ which, value }) => ({
+    color: age(value),
+    label: which === 'age' ? 'created' : 'updated',
+    message: formatDate(value),
+  }),
+}
```

Comment:
In the future it might be nice to design a mini-framework that can live inside of a service and deal with child-service logic that crosses boundaries of fetch (sometimes), schema, transform, and render. I know we've got these sprinkled about and it might be nice to design a single way to guide these implementations.